src/text-editor.coffee

```diff
@@ -176,9 +176,15 @@ class TextEditor extends Model
       @subscribe @displayBuffer.onDidAddDecoration (decoration) => @emit 'decoration-added', decoration
       @subscribe @displayBuffer.onDidRemoveDecoration (decoration) => @emit 'decoration-removed', decoration
 
+  subscribeToTabTypeConfig: ->
+    @tabTypeSubscription?.dispose()
```

Comment:
This is not super pretty. As I need to either add it to @disposables then remove on a grammar change, or explicitly destroy it in the `destroyed` method. I chose the latter